hack/verify-codegen.sh

```diff
@@ -23,28 +19,18 @@ cleanup() {
 }
 trap "cleanup" EXIT SIGINT
 
-echo ">> Temporary output directory ${TEMP_DIR}"
-
-# Ensure we can execute.
-chmod +x ${CODEGEN_PKG}/generate-groups.sh
-
-${CODEGEN_PKG}/generate-groups.sh "client" \
-    github.com/fluent/fluent-operator/v2/apis/generated github.com/fluent/fluent-operator/v2/apis \
-    "fluentbit:v1alpha2 fluentd:v1alpha1" \
-    --output-base "${TEMP_DIR}" \
-    --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.go.txt
-
-TMP_DIFFROOT=${TEMP_DIR}/github.com/fluent/fluent-operator/v2/apis/generated/clientset
 DIFFROOT=${SCRIPT_ROOT}/apis/generated/clientset
+cp -a "${DIFFROOT}"/* "${TEMP_DIR}"
 
-echo "diffing ${DIFFROOT} against freshly generated clientset"
+"${SCRIPT_ROOT}/hack/update-codegen.sh"
+echo "diffing ${DIFFROOT} against freshly generated codegen"
 ret=0
-diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
-cp -a "${TMP_DIFFROOT}"/* "${DIFFROOT}"
+diff -Naupr "${DIFFROOT}" "${TEMP_DIR}" || ret=$?
+cp -a "${TEMP_DIR}"/* "${DIFFROOT}"
 if [[ $ret -eq 0 ]]
 then
   echo "${DIFFROOT} up to date."
 else
-  echo "${DIFFROOT} is out of date. Please rerun make generate"
+  echo "${DIFFROOT} is out of date. Please run hack/update-codegen.sh"
```

Comment:
This had to be changed because `generate-groups.sh` seems to delete the tree now and thus the comparison was wrong.

This change essentially copies the current tree into a temp dir, then generates the clients "as normal" (full on replacing the tree) and then diffs the copied out tree with the now regenerated clients. Has the added benefit, that the actual generation is deduplicated.